Competition2017/src/competition/subsystems/drive/commands/AutonomousDriveAlongCurveCommand.java

```diff
@@ -11,69 +11,79 @@
 import xbot.common.properties.XPropertyManager;
 
-public class DriveAlongCurveCommand extends BaseDriveCommand {
+public class AutonomousDriveAlongCurveCommand extends BaseDriveCommand {
     
-    private final DoubleProperty totalTime;
+    private final DoubleProperty totalTimeOfAutoCurve;
     private final DoubleProperty timeStep;
     private final DoubleProperty robotTrackWidth;
     
-    private final double initialCommandStart;
-    private int i;
+    private double[][] waypoints;
     
-    AutonomousCurveTrajectoryPlanner path;
+    private final double initialCommandTime;
+    private int indexOfTimeStep;
+    
+    final AutonomousCurveTrajectoryPlanner path;
     
     @Inject
-    public DriveAlongCurveCommand
+    public AutonomousDriveAlongCurveCommand
             (DriveSubsystem driveSubsystem,
             XPropertyManager prop
             ) {
         
         super(driveSubsystem);
         this.requires(driveSubsystem);
         
-        this.i = 0;
-        this.initialCommandStart = Timer.getFPGATimestamp();
+        this.indexOfTimeStep = 0;
+        this.initialCommandTime = Timer.getFPGATimestamp();
         
         //Input points to create trajectory for robot
-        double[][] waypoints = new double[][]{
+         this.waypoints = new double[][]{
             {1,1},
             {2,3},
             {4,5},
             {5,4},
             {7,7}
         };
         
-        totalTime = prop.createPersistentProperty("Total time for autonomous command", 10);//seconds
+        totalTimeOfAutoCurve = prop.createPersistentProperty("Total time for autonomous curve command", 10);//seconds
         timeStep = prop.createPersistentProperty("Rate at which controller runs", 0.1);//rate at which rio iterates, seconds
-        robotTrackWidth = prop.createPersistentProperty("Distance between left and right wheel", 2); //feet
+        robotTrackWidth = prop.createPersistentProperty("Distance between left and right wheel", 2); //inches
         
-        AutonomousCurveTrajectoryPlanner path = new AutonomousCurveTrajectoryPlanner(waypoints);
+        path = new AutonomousCurveTrajectoryPlanner(waypoints);
+    }
+    
+    public void setWayPoints(double[][] waypoints){
+        this.waypoints = waypoints;
+    }
+    
+    public AutonomousCurveTrajectoryPlanner getPath(){
+        return path;
     }
     
     @Override
     public void initialize(){
-        path.calculate(totalTime.get(), timeStep.get(), robotTrackWidth.get());
+        path.calculate(totalTimeOfAutoCurve.get(), timeStep.get(), robotTrackWidth.get());
     }
     
     @Override
     public void execute(){
-        if(Timer.getFPGATimestamp()-initialCommandStart == path.smoothLeftVelocity[i][0]){
-            driveSubsystem.tankDriveVelocityPid(path.smoothLeftVelocity[i][1], path.smoothRightVelocity[i][1]);
-            i++;
+        if(Timer.getFPGATimestamp()-initialCommandTime >= path.smoothLeftVelocity[indexOfTimeStep][0]){
```

Comment:
This will ensure that the points are always executed a bit later than they should, which will add up with lots of points. Not sure if there's a better way in the immediate future though.